src/components/UniversalAmount.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+    <span class="amount" :class="{ approx: showApprox && isApprox }">
+        {{ formattedAmount }}
+        <span class="currency" :class="currency">{{currency}}</span>
+    </span>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import bigInt from 'big-integer';
+
+@Component
+export default class UniversalAmount extends Vue {
+    /**
+     * Amount in smallest unit
+     */
+    @Prop({type: [Number, Object]}) public amount!: number | bigInt.BigInteger;
+    /**
+     * Actual decimal count of the currency being displayed
```

Comment:
That's different semantics than the `decimals` in `Amount.vue`.
I therefore think, it should have a different name.